servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -291,6 +292,27 @@ void RendererCanvasCull::_cull_canvas_item(Item *p_canvas_item, const Transform2
 	final_xform = parent_xform * final_xform;
 
 	Rect2 global_rect = final_xform.xform(rect);
+	if (repeat_source_item && (repeat_size.x || repeat_size.y)) {
+		Size2 repeat_size_x_offset = repeat_source_item->final_transform.basis_xform(Size2(repeat_size.x, 0));
+		Size2 repeat_size_y_offset = repeat_source_item->final_transform.basis_xform(Size2(0, repeat_size.y));
+
+		// Top-left repeated rect.
+		Rect2 corner_rect = global_rect;
+		corner_rect.position -= (repeat_times / 2) * (repeat_size_x_offset + repeat_size_y_offset);
+		global_rect = corner_rect;
+
```

Comment:
You need to multiply the `repeat_times` to the offsets for the other corners.

Reply:
Oh, right, simplified it too much while refactoring a little. :upside_down_face: Nice catch!

Actually, instead of multiplying by `repeat_times` here I'll revert back to the previous version with 3 `basis_xform()`s (instead of 2 like here), as multiplying the results would be a little more floating point error prone (and `basis_xform` is cheap). It again likely doesn't matter though. :slightly_smiling_face: